tests/integration/test_interface.py

```diff
@@ -65,54 +64,54 @@ def test_generate_dataset_from_multiple_shards(
-    # full data size (we would need unnoised data for that), so we just check
-    # for similar lengths
-    assert 0.9 <= (len(noised_dataset) / len(noised_sample)) <= 1.1
-    # Check that columns are identical
-    assert noised_dataset.columns.equals(noised_sample.columns)
-
-    # Check that each columns level of noising are similar
-    check_noised_dataset, check_original_dataset, shared_dataset_idx = _get_common_datasets(
-        original, noised_dataset
-    )
+        multi_shard_noised_data = multi_shard_noised_data.compute()
 
-    config_tree = get_configuration(config)
-    for col_name in check_noised_dataset.columns:
-        col = COLUMNS.get_column(col_name)
-        if col.noise_types:
-            noise_level_dataset, to_compare_dataset_idx = _get_column_noise_level(
-                column=col,
-                noised_data=check_noised_dataset,
-                unnoised_data=check_original_dataset,
-                common_idx=shared_dataset_idx,
-            )
-
-            # Validate noise for each data object
-            validate_column_noise_level(
-                dataset_name=dataset_name,
-                check_data=check_original_dataset,
-                check_idx=to_compare_dataset_idx,
-                noise_level=noise_level_dataset,
-                col=col,
-                config=config_tree,
-                fuzzy_name="test_generate_dataset_from_sample_and_source_dataset",
-                validator=fuzzy_checker,
-            )
+    assert multi_shard_noised_data.columns.equals(single_shard_noised_data.columns)
+
+    # This index handling is adapted from _get_common_datasets
+    # in integration/conftest.py
+    # Define indexes
+    idx_cols = IDX_COLS.get(unnoised_dataset.dataset_schema.name)
+    unnoised_dataset._reformat_dates_for_noising()
+    unnoised_dataset.data = coerce_dtypes(
+        unnoised_dataset.data, unnoised_dataset.dataset_schema
+    )
+    check_original = unnoised_dataset.data.set_index(idx_cols)
+    check_single_noised = single_shard_noised_data.set_index(idx_cols)
+    check_multi_noised = multi_shard_noised_data.set_index(idx_cols)
+
+    # Ensure the idx_cols are unique
+    assert check_original.index.duplicated().sum() == 0
+    assert check_single_noised.index.duplicated().sum() == 0
+    assert check_multi_noised.index.duplicated().sum() == 0
+
+    # Get shared indexes
+    shared_idx = pd.Index(
+        set(check_original.index)
+        .intersection(set(check_single_noised.index))
+        .intersection(set(check_multi_noised.index))
+    )
+    check_original = check_original.loc[shared_idx]
+    check_single_noised = check_single_noised.loc[shared_idx]
+    check_multi_noised = check_multi_noised.loc[shared_idx]
+
+    for col in check_single_noised.columns:
+        fuzzy_checker.fuzzy_assert_proportion(
+            target_proportion=(check_single_noised[col] != check_original[col]).mean(),
```

Comment:
I don't think this is _exactly_ correct, but I'll allow it 😄Assuming the datasets are reasonably big it should work

Reply:
Haha yup the engineers discussed that is not how this is intended to be used, but good to hear that it's probably good enough :)